File: ios/Porchivo/Models/IncidentKind.swift
//
//  IncidentKind.swift
//  Porchivo
//
//  Incident types mirroring the DB `incident_type` enum on `incident_reports`.
//  Mirrors the Expo app's file-incident screen.
//

import Foundation

enum IncidentKind: String, CaseIterable, Identifiable {
    case missingPackage = "missing_package"
    case deliveredNotFound = "delivered_not_found"
    case misdelivered
    case damaged
    case tampered
    case suspiciousActivity = "suspicious_activity"
    case heldTooLong = "held_too_long"
    case wrongPickup = "wrong_pickup"
    case ruleViolation = "rule_violation"
    case carrierFailure = "carrier_failure"
    case duplicateComplaint = "duplicate_complaint"
    case other

    var id: String { rawValue }

    var label: String {
        switch self {
        case .missingPackage: return "Package missing"
        case .deliveredNotFound: return "Delivered but not found"
        case .misdelivered: return "Delivered to wrong unit"
        case .damaged: return "Package damaged"
        case .tampered: return "Opened or tampered with"
        case .suspiciousActivity: return "Suspicious activity"
        case .heldTooLong: return "Held too long"
        case .wrongPickup: return "Wrong person picked up"
        case .ruleViolation: return "Outside delivery rules"
        case .carrierFailure: return "Carrier didn't follow instructions"
        case .duplicateComplaint: return "Duplicate complaint"
        case .other: return "Something else"
        }
    }

    var emoji: String {
        switch self {
        case .missingPackage: return "📦"
        case .deliveredNotFound: return "🔍"
        case .misdelivered: return "📬"
        case .damaged: return "💥"
        case .tampered: return "🔓"
        case .suspiciousActivity: return "🚨"
        case .heldTooLong: return "⏳"
        case .wrongPickup: return "🙅"
        case .ruleViolation: return "📋"
        case .carrierFailure: return "🚚"
        case .duplicateComplaint: return "🔁"
        case .other: return "❓"
        }
    }

    /// Default one-line title, pre-filled when the user picks a type.
    var defaultTitle: String {
        switch self {
        case .missingPackage: return "Package missing from expected location"
        case .deliveredNotFound: return "Carrier marked delivered but package not found"
        case .misdelivered: return "Package delivered to wrong unit"
        case .damaged: return "Package arrived damaged"
        case .tampered: return "Package appears opened or tampered with"
        case .suspiciousActivity: return "Suspicious activity near delivery area"
        case .heldTooLong: return "Package held in common area too long"
        case .wrongPickup: return "Package picked up by wrong person"
        case .ruleViolation: return "Delivery made outside community rules"
        case .carrierFailure: return "Carrier failed to follow delivery instructions"
        case .duplicateComplaint: return "Duplicate complaint about same package"
        case .other: return "Package delivery issue"
        }
    }

    /// Package-issue types where the carrier handles claims — shows the
    /// carrier reminder and the estimated-value field.
    var isCarrierAction: Bool {
        switch self {
        case .missingPackage, .deliveredNotFound, .misdelivered, .damaged, .tampered, .carrierFailure:
            return true
        case .suspiciousActivity, .heldTooLong, .wrongPickup, .ruleViolation, .duplicateComplaint, .other:
            return false
        }
    }
}

File: android/app/src/main/java/com/rork/porchivo/model/IncidentKind.kt
package com.rork.porchivo.model

/**
 * Incident types mirroring the DB `incident_type` enum on `incident_reports`.
 * Mirrors the Expo app's file-incident screen.
 */
enum class IncidentKind(
    val value: String,
    val label: String,
    val emoji: String,
    /** Package-issue types where the carrier handles claims — shows the carrier reminder and the estimated-value field. */
    val isCarrierAction: Boolean,
) {
    MISSING_PACKAGE("missing_package", "Package missing", "📦", true),
    DELIVERED_NOT_FOUND("delivered_not_found", "Delivered but not found", "🔍", true),
    MISDELIVERED("misdelivered", "Delivered to wrong unit", "📬", true),
    DAMAGED("damaged", "Package damaged", "💥", true),
    TAMPERED("tampered", "Opened or tampered with", "🔓", true),
    SUSPICIOUS_ACTIVITY("suspicious_activity", "Suspicious activity", "🚨", false),
    HELD_TOO_LONG("held_too_long", "Held too long", "⏳", false),
    WRONG_PICKUP("wrong_pickup", "Wrong person picked up", "🙅", false),
    RULE_VIOLATION("rule_violation", "Outside delivery rules", "📋", false),
    CARRIER_FAILURE("carrier_failure", "Carrier didn't follow instructions", "🚚", true),
    DUPLICATE_COMPLAINT("duplicate_complaint", "Duplicate complaint", "🔁", false),
    OTHER("other", "Something else", "❓", false);

    /** Default one-line title, pre-filled when the user picks a type. */
    val defaultTitle: String
        get() = when (this) {
            MISSING_PACKAGE -> "Package missing from expected location"
            DELIVERED_NOT_FOUND -> "Carrier marked delivered but package not found"
            MISDELIVERED -> "Package delivered to wrong unit"
            DAMAGED -> "Package arrived damaged"
            TAMPERED -> "Package appears opened or tampered with"
            SUSPICIOUS_ACTIVITY -> "Suspicious activity near delivery area"
            HELD_TOO_LONG -> "Package held in common area too long"
            WRONG_PICKUP -> "Package picked up by wrong person"
            RULE_VIOLATION -> "Delivery made outside community rules"
            CARRIER_FAILURE -> "Carrier failed to follow delivery instructions"
            DUPLICATE_COMPLAINT -> "Duplicate complaint about same package"
            OTHER -> "Package delivery issue"
        }
}
